src/components/DataTable/DataTable.tsx

```diff
@@ -28,33 +17,33 @@ import {
   TableHeader,
   TableRow,
   ViewDialog,
-  ContactRejectDialogProps,
-  RevokeKeyDialogProps,
-  TransactionEditDialogProps,
-  DestinationEditDialogProps,
 } from '@/components';
+import { isContact } from '@/utils';
+import { AccessKey, Contact, Destination, PaymailAddress, Tx, XPub } from '@bsv/spv-wallet-js-client';
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getPaginationRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+import { EllipsisVertical } from 'lucide-react';
+
+import React, { useState } from 'react';
+
+export type RowType = XPub | Contact | AccessKey | Destination | PaymailAddress | Tx;
 
 interface DataTableProps<TData, TValue> {
   columns: ColumnDef<TData, TValue>[];
   data: TData[];
-  EditDialog?: React.ComponentType<ContactEditDialogProps | TransactionEditDialogProps | DestinationEditDialogProps>;
-  AcceptDialog?: React.ComponentType<ContactAcceptDialogProps>;
-  DeleteDialog?: React.ComponentType<ContactDeleteDialogProps>;
-  RejectDialog?: React.ComponentType<ContactRejectDialogProps>;
-  RevokeKeyDialog?: React.ComponentType<RevokeKeyDialogProps>;
+  renderItem?: (row: Row<RowType>) => React.ReactNode;
```

Comment:
Use generics from the interface not some custom type, this way you won't need to use type assertions later in code.
```suggestion
  renderItem?: (row: Row<TData>) => React.ReactNode;
```